File: 0201_0300/0287/0287.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m = (int(x) for x in f.readline().split())
    s = f.readline().strip()
    l = set(s[i : i + m] for i in range(n - m + 1))
    q.write(str(len(l)))

File: 0201_0300/0287/0287.c
#include <stdio.h>
#include <string.h>

char *find(char *s1, char *s2, int len)
{
	while (strncmp(s1, s2, len))
		++s1;
	return (s1);
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, m, k = 0;
	char s[101], *t;

	fscanf(f, "%d%d%s", &n, &m, s);
	t = s + n - m + 1;
	while (--t >= s)
		if (find(s, t, m) == t)
			++k;
	fprintf(q, "%d", k);
	return 0;
}
